demisto_sdk/commands/content_graph/objects/base_content.py

```diff
@@ -1,14 +1,44 @@
 import json
-from abc import ABC
-from typing import Any, Dict, List
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Type, cast
 
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, DirectoryPath, Field
+from pydantic.main import ModelMetaclass
 
 from demisto_sdk.commands.common.constants import MarketplaceVersions
 from demisto_sdk.commands.content_graph.common import ContentType
 
+content_type_to_model: Dict[ContentType, Type['BaseContent']] = {}
 
-class BaseContent(ABC, BaseModel):
+
+class ContentModelMetaclass(ModelMetaclass):
+    def __new__(cls, name, bases, namespace, content_type: ContentType = None, **kwargs):
```

Comment:
Please check whether it's possible to implement the same functionality without explicitly providing `content_type` (e.g. getting it from the class name)